test/unit/test_ranges.py

```diff
@@ -0,0 +1,223 @@
+            device=acc_node['device'],
+            partition=partition,
+            account=account,
+        )
+        acc_headers = {'X-Timestamp': normalize_timestamp(time.time())}
+        acc_conn = HTTPConnection(acc_node['ip'], acc_node['port'])
+        acc_conn.request('PUT', path, headers=acc_headers)
+        r = acc_conn.getresponse()
+        self.assertEqual(r.status, 201)
+
+        # Create test container
+        self.proxy_host, self.proxy_port = proxy_socket.getsockname()
+        proxy_conn = HTTPConnection(self.proxy_host, self.proxy_port)
+        proxy_conn.request('PUT', '/v1/a/c/')
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 201)
+
+    def test_ranges(self):
+        self.check_path = '%s/a/c/o' % self.sproxyd_path
+        self.check_method = 'GET'
+        self.response = '_' * 100
+        object_path = '/v1/a/c/o'
+        proxy_conn = HTTPConnection(self.proxy_host, self.proxy_port)
+
+        # Get last 95 bytes
+        headers = {'RANGE': 'bytes=5-'}
+        self.expected_headers = {'RANGE': 'bytes=5-99'}
+        proxy_conn.request('GET', object_path, headers=headers)
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 206)
+        response_headers = dict(r.getheaders())
+        r.read()
+        self.assertEqual(response_headers['content-range'], 'bytes 5-99/100')
+
+        # Get last 5 bytes
+        headers = {'RANGE': 'bytes=-5'}
+        self.expected_headers = {'RANGE': 'bytes=95-99'}
+        proxy_conn.request('GET', object_path, headers=headers)
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 206)
+        response_headers = dict(r.getheaders())
+        r.read()
+        self.assertEqual(response_headers['content-range'], 'bytes 95-99/100')
+
+        # Get first 5 bytes
+        headers = {'RANGE': 'bytes=0-4'}
+        self.expected_headers = headers
+        proxy_conn.request('GET', object_path, headers=headers)
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 206)
+        response_headers = dict(r.getheaders())
+        r.read()
+        self.assertEqual(response_headers['content-range'], 'bytes 0-4/100')
+
+        # Get 20 intermediate bytes
+        headers = {'RANGE': 'bytes=20-40'}
+        self.expected_headers = headers
+        proxy_conn.request('GET', object_path, headers=headers)
+        r = proxy_conn.getresponse()
+        self.assertEqual(r.status, 206)
+        response_headers = dict(r.getheaders())
+        r.read()
+        self.assertEqual(response_headers['content-range'], 'bytes 20-40/100')
+
+    def validate_request_headers(self, method, path, headers):
+        """
+        Validate headers of requests for an entity matching `check_path`
+        with method `check_method`.
+        """
+        if (self.check_method == method and self.check_path == path):
+            for header in self.expected_headers:
+                self.assertEqual(headers[header],
```

Comment:
This assertion runs inside the sproxyd mock, which runs in a greenlet. Does this work as intended?